graphql/query_resolver.go

```diff
@@ -807,9 +808,29 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 	if len(requesters) == 0 {
 		requesters = evergreen.SystemVersionRequesterTypes
 	}
+
+	skipOrderNumber := utility.FromIntPtr(options.SkipOrderNumber)
+	revision := utility.FromStringPtr(options.Revision)
+
+	if options.SkipOrderNumber == nil && revision != "" {
+		if len(revision) < 7 {
+			graphql.AddError(ctx, PartialError.Send(ctx, fmt.Sprintf("at least 7 characters must be provided for the revision")))
+		} else {
+			found, err := model.VersionFindOne(model.VersionByProjectIdAndRevisionPrefix(projectId, revision))
+			if err != nil {
+				graphql.AddError(ctx, PartialError.Send(ctx, fmt.Sprintf("could not find git commit '%s'", revision)))
+			} else if found == nil {
+				graphql.AddError(ctx, PartialError.Send(ctx, fmt.Sprintf("could not find git commit '%s'", revision)))
+			} else {
+				// Offset the order number so the specified revision lands nearer to the center of the page.
+				skipOrderNumber = found.RevisionOrderNumber + 1 + limit/2
```

Comment:
Just confirming that only the limit is meant to be halved here, not the entire sum?

Reply:
yep, the limit actually represents how many commits a user is currently viewing on the page. By adding half of the limit, it means that the commit will end up roughly in the center of the page